### ISSUE TITLE
WSL over tunnel

### ISSUE THREAD

**Issue by caiweims** [microsoft/vscode-remote-release#8685]
Type: <b>Bug</b>

Error running command remote-explorer.connectToFolderInCurrentWindow: Cannot read properties of undefined (reading 'env'). This is likely caused by the extension that contributes remote-explorer.connectToFolderInCurrentWindow.

Extension version: 0.4.1
VS Code version: Code 1.80.0 (660393deaaa6d1996740ff4880f1bad43768c814, 2023-07-04T15:06:02.407Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6300U CPU @ 2.40GHz (4 x 2496)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|3.92GB (0.53GB free)|
|Process Argv|--crash-reporter-id 0cfb29ed-15b7-4042-960d-1b25261afad5|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30410667
py29gd2263:30776702
vscaac:30438847
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
2e4cg342:30602488
pyind779:30671433
f6dab269:30613381
pythonsymbol12:30671437
2i9eh265:30646982
showlangstatbar:30737416
vsctsb:30748421
pythonms35:30701012
a2ce3375:30757347
pythonfmttext:30731395
pythoncmvfstrcf:30756944
fixshowwlkth:30771522
showindicator:30766890
pythongtdpath:30769146
i26e3531:30780429
pythonnosm12tcf:30779713
pythonidxpt:30784022
pythonnocebcf:30776496
h7j2d465:30772216

```

</details>

**Comment by tobico** [comment]
I'm getting the same error when I try to connect to tunneled WSL.

If I first connect to the remote tunnel and then attempt to switch to WSL, I get this error instead:

> Error running command remote-explorer.connectToFolderInCurrentWindow: command 'remote-wsl.connect' not found. This is likely caused by the extension that contributes remote-explorer.connectToFolderInCurrentWindow.

**Comment by connor4312** [comment]
Please make sure the _tunnel_ is also up to date with VS Code 1.80. If you're hosting the tunnel from the UI, head over to `Help > Check for Updates`. If you're using the standalone CLI, run `code update`.

**Comment by ypolosov** [comment]
i have the same error ...both sides are updated

**Comment by connor4312** [comment]
Could you please run the "toggle developer tools" command and check the console for any errors?

**Comment by connor4312** [comment]
@ypolosov what is the exact error you're getting?

Also, could you preemptively run the "toggle developer tools" command and check the console for errors?

**Comment by Theelx** [comment]
Update:
Fixed it by changing from the tab in the first picture to the second, then clicking connect on Ubuntu-22.04.
![Screenshot from 2023-07-10 14-29-13](https://github.com/microsoft/vscode-remote-release/assets/43764914/e2621afe-9d18-4cb0-89ba-31a22cbbbe54)
-->
![Screenshot from 2023-07-10 14-29-24](https://github.com/microsoft/vscode-remote-release/assets/43764914/b8cf85fd-0f3e-4085-b8ee-5de06ac59786)

**Comment by eladts** [comment]
@Theelx I don't see the _WSL Targets_ tab on the remote machine (using vscode.dev), so your workaround does not work for me. 

@connor4312 What is the URL format for WSL over tunnels? Perhaps I can reach WSL through a Windows hosted tunnel by manually constructing the URL.

**Comment by pushad** [comment]
I'm having the same issue. VS Code running on W11, connecting using Tunnel via vscode.dev and getting the same error as issue. WSL Targets not visible in remote explorer dropdown.

**Comment by connor4312** [comment]
This should be fixed in the just-released version of remote tunnels (v1.3.0). Users on vscode.dev may need to refresh the page to get the update.

If you hit the "not found" issue, please check and make sure your version of the WSL extension is also up to date (v0.80.2).
